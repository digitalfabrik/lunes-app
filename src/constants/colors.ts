export const COLORS = {
  lunesBlack: '#000e38',
  lunesWhite: '#f9fafb',
  lunesGreyMedium: '#5b616a',
  lunesGreyDark: '#3a4049',
  white: '#ffffff',
  lunesBlackUltralight: '#e0e4ed',
  lunesFunctionalIncorrectDark: '#ff5252',
  lunesRedLight: '#f57f7a',
  lunesGreenLight: '#7ae6d8',
  lunesGreenMedium: '#20c5b4',
  lunesBlueLight: '#97aef1',
  lunesOrangeLight: '#ffbb4a',
  lunesArtikelDas: '#72f399',
  lunesArtikelDiePlural: '#eee12d',
  lunesArtikelDie: '#faa7a7',
  lunesArtikelDer: '#8cc8f3',
  lunesOverlay: 'rgba(0, 14, 56, 0.8)',
  lunesBlackLight: '#777e91',
  blue: '#0000FF',
  lunesFunctionalCorrectDark: '#0adb75',
  lunesFunctionalAlmostCorrectDark: '#ffbb4a',
  lunesRedDark: '#F1635F',
  lunesRed: '#ffa3a3',
  shadow: 'rgba(0, 0, 0, 0.6)',
  lunesBlackMedium: '#3d4662'
}
