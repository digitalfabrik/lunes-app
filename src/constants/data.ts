import {
  easy,
  hard,
  CorrectIcon,
  IncorrectIcon,
  AlmostCorrectIcon,
  CorrectEntriesIcon,
  IncorrectEntriesIcon,
  AlmostCorrectEntriesIcon
} from '../../assets/images'

export const SCREENS = {
  profession: 'Profession',
  professionSubcategory: 'ProfessionSubcategory',
  exercises: 'Exercises',
  vocabularyOverview: 'VocabularyOverview',
  vocabularyTrainer: 'VocabularyTrainer',
  initialSummaryScreen: 'InitialSummary',
  ResultsOverview: 'ResultsOverview',
  CorrectResults: 'CorrectResults',
  ResultScreen: 'ResultScreen',
  AlmostCorrectResults: 'AlmostCorrectResults',
  IncorrectResults: 'IncorrectResults'
}

export const RESULT_TYPE = ['correct', 'similar', 'incorrect']

export const EXERCISES = [
  {
    id: 1,
    title: 'Vocabulary Overview',
    description: 'All Words',
    Level: easy,
    nextScreen: SCREENS.vocabularyOverview
  },
  {
    id: 2,
    title: 'Vocabulary Trainer',
    description: 'Write words with articles',
    Level: hard,
    nextScreen: SCREENS.vocabularyTrainer
  }
]

export const BUTTONS_THEME = {
  light: 'light',
  dark: 'dark'
}

export const RESULTS = [
  {
    id: 1,
    title: 'Correct entries',
    icon: CorrectIcon,
    nextScreen: SCREENS.CorrectResults
  },
  {
    id: 2,
    title: 'Almost correct entries',
    icon: AlmostCorrectIcon,
    nextScreen: SCREENS.AlmostCorrectResults
  },
  {
    id: 3,
    title: 'Incorrect entries',
    icon: IncorrectIcon,
    nextScreen: SCREENS.IncorrectResults
  }
]

export const ARTICLES = {
  die: 'die',
  der: 'der',
  das: 'das',
  diePlural: 'die (plural)'
}

export const RESULT_PRESETS = {
  similar: {
    Icon: AlmostCorrectEntriesIcon,
    title: 'Almost Correct',
    next: { title: 'INCORRECT', type: 'incorrect' }
  },
  correct: {
    Icon: CorrectEntriesIcon,
    title: 'Correct',
    next: { title: 'ALMOST CORRECT', type: 'similar' }
  },
  incorrect: {
    Icon: IncorrectEntriesIcon,
    title: 'Incorrect',
    next: { title: 'CORRECT', type: 'correct' }
  }
}
