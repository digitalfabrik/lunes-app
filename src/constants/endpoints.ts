export const ENDPOINTS = {
  professions: {
    all: '/disciplines'
  },
  subCategories: {
    all: '/training_set/:id'
  },
  documents: {
    all: '/documents/:id'
  }
}
