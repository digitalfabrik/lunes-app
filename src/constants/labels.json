{
  "scopeSelection": {
    "welcome": "Willkommen bei Lunes!",
    "selectProfession": "Wähle den Beruf, den du lernen möchtest.",
    "skipSelection": "Überspringen",
    "confirmSelection": "Starten"
  },
  "home": {
    "welcome": "Willkommen bei Lunes!",
    "haveFun": "Viel Spaß beim Lernen",
    "progressDescription": "Modulen erledigt",
    "start": "Starten",
    "continue": "Fortsetzen",
    "viewDisciplines": "Module ansehen",
    "addCustomDiscipline": "Bereich hinzufügen",
    "customDisciplineSection": "Individueller Lernbereich",
    "customDisciplineExplanation": "Hier kannst du eigene Lernbereiche hinzufügen",
    "errorLoadCustomDiscipline": "Fehler: Bereich konnte nicht geladen werden. Ungültiger Schlüssel:",
    "deleteProfession": "Bereich löschen",
    "impressum": "Impressum"
  },
  "manageSelection": {
    "heading": "Verwalte deine Lernbereiche",
    "yourProfessions": "Deine Berufsauswahl",
    "yourCustomDisciplines": "Deine individuellen Bereiche",
    "yourSavedProgress": "Deine gespeicherten Lernbereiche",
    "addProfession": "Beruf hinzufügen",
    "descriptionAddCustomDiscipline": "Hier kannst du eigene Lernbereiche mit deinem QR-Code hinzufügen.",
    "deleteModal": {
      "confirmationQuestion": "Willst du den Bereich wirklich löschen?",
      "explanation": "Deine Lernfortschritte werden gespeichert und du kannst den Beruf später wiederherstellen.",
      "confirm": "Ja, löschen!"
    }
  },
  "addCustomDiscipline": {
    "heading": "Individuellen Bereich hinzufügen",
    "description": "Bitte gib den Code über die Tastatur ein oder scanne den QR-Code.",
    "placeholder": "Code eingeben",
    "submitLabel": "hinzufügen",
    "backNavigation": "Zur Startseite",
    "error": {
      "alreadyAdded": "Dieser Code wurde bereits erfolgreich hinzufügt.",
      "wrongCode": "Dieser Code existiert nicht (ungültiger Code).",
      "technical": "Ein technischer Fehler ist aufgetreten. Prüfen Sie Ihre Internetverbindung."
    }
  },
  "exercises": {
    "vocabularyList": {
      "title": "Wortliste",
      "description": "Alle Wörter kennenlernen",
      "alternativeWords": "Andere Wörter",
      "suggestAlternative": "Anderes Wort vorschlagen"
    },
    "wordChoice": {
      "title": "Level 1",
      "description": "Das richtige Wort auswählen",
      "errorWrongDisciplineSize": "Die Übung ist für dieses Modul nicht möglich (zu wenig Wörter)"
    },
    "articleChoice": {
      "title": "Level 2",
      "description": "Den richtigen Artikel auswählen"
    },
    "lockedExerciseModal": {
      "title": "Hoppla, dieses Level ist noch nicht freigeschalten.",
      "confirmButtonLabel": "Zu",
      "confirmButtonLabelDeclinated": "Zur",
      "descriptionPart1": "Du musst",
      "descriptionPart2": "abschließen, um das nächste Level freizuschalten."
    },
    "write": {
      "title": "Level 3",
      "description": "Das Wort mit Artikel schreiben",
      "checkInput": "Eingabe überprüfen",
      "showSolution": "Lösung anzeigen",
      "insertAnswer": "Wort mit Artikel eingeben",
      "feedback": {
        "articleMissing": "Ups, deine Eingabe ist nicht vollständig. Füge einen Artikel ein.",
        "wrongWithSolution": "Die richtige Antwort ist:",
        "wrong": "Das war leider nicht richtig.",
        "almostCorrect1": "Deine Eingabe",
        "almostCorrect2": "ist fast richtig.",
        "correct": "Toll, weiter so! \nDeine Eingabe ist richtig."
      }
    },
    "tryLater": "Später versuchen",
    "cheat": {
      "succeed": "Solve All",
      "fail": "Fail All"
    },
    "next": "Nächstes Wort",
    "showResults": "Ergebnisse anzeigen",
    "cancelModal": {
      "cancelAsk": "Möchtest du diese Übung wirklich beenden?",
      "cancel": "Beenden"
    },
    "feedback": {
      "positive": "Weiter so, das war super!",
      "negative": "Das geht besser. Probier es nochmal."
    }
  },
  "results": {
    "feedbackGood": "Toll, weiter so! \nDu hast die Übung sehr gut gemacht.",
    "feedbackBad": "Nicht aufgeben! \nVersuche es noch einmal!",
    "finishedDiscipline": "Super, du hast das Modul abgeschlossen",
    "unlockExercise": {
      "part1": "Glückwunsch",
      "part2": "ist jetzt freigeschaltet."
    },
    "correct": "richtig",
    "entries": "Eingaben",
    "action": {
      "nextExercise": "Nächste Übung",
      "continue": "Weitermachen",
      "repeat": "Übung wiederholen",
      "disciplineOverview": "Zur Modulübersicht",
      "backToWordlist": "Zurück zur Wortliste"
    },
    "of": "von",
    "share": {
      "description": "Teile deine Ergebnisse mit deinen Freunden!",
      "button": "Ergebnisse teilen",
      "message1": "Meine Ergebnisse für",
      "message2": "in Lunes:",
      "message3": "Wörter richtig. Und du?"
    }
  },
  "feedback": {
    "sendFeedback": "Senden",
    "question": "Hast du Feedback oder einen Wortvorschlag?",
    "mailPlaceholder": "Deine Email für Fragen (optional)",
    "feedbackPlaceholder": "Teile dein Feedback hier"
  },
  "general": {
    "home": "Start",
    "favorites": "Favoriten",
    "dictionary": "Lexikon",
    "userVocabulary": "Meine Vokabeln",
    "header": {
      "overview": "Übersicht",
      "overviewExercises": "Übersicht der Übungen",
      "cancelExercise": "Übung beenden",
      "manageSelection": "Bereiche verwalten",
      "settings": "Einstellungen",
      "wordFeedback": "Feedback geben",
      "overflowMenu": "Mehr Optionen"
    },
    "back": "Zurück",
    "plurals": "Plural",
    "noResults": "Es wurden leider keine Ergebnisse gefunden.",
    "error": {
      "unknown": "Ein unbekannter Fehler ist aufgetreten.",
      "noWifi": "Es konnte keine Verbindung mit dem Internet hergestellt werden.",
      "retryButton": "Erneut versuchen",
      "somethingWentWrong": "Ups, da ist etwas schief gelaufen!"
    },
    "rootDiscipline": "Beruf",
    "rootDisciplines": "Berufe",
    "discipline": "Modul",
    "disciplines": "Module",
    "word": "Wort",
    "words": "Wörter",
    "camera": {
      "noAuthorization": {
        "description": "Kamerazugriff erforderlich. Öffnen Sie die Einstellungen, tippen Sie auf Berechtigungen und tippen Sie auf Erlauben.",
        "settings": "Einstellungen"
      }
    }
  },
  "favorites": "Favoriten",
  "dictionary": {
    "noResults": "Es wurden leider keine Ergebnisse gefunden."
  },
  "search": {
    "enterWord": "Wort eingeben"
  },
  "settings": {
    "settings": "Einstellungen",
    "appStability": "App Stabilität",
    "appStabilityExplanation": "Durch das automatische Senden von Fehlerberichten die Lunes App verbessern",
    "version": "Version",
    "debugModal": {
      "sentry": "Send Sentry Error",
      "currentCMS": "current cms",
      "changeCMS": "change cms",
      "enableDevMode": "Enable DevMode",
      "disableDevMode": "Disable DevMode"
    }
  },
  "userVocabulary": {
    "myWords": "Meine Wörter",
    "create": "Eigenes Wort erstellen",
    "overview": {
      "list": "Vokabelliste",
      "create": "Vokabeln erstellen",
      "practice": "Vokabeln üben"
    },
    "creation": {
      "headline": "Vokabel erstellen",
      "wordPlaceholder": "Wort eingeben*",
      "articlePlaceholder": "Artikel auswählen*",
      "saveButton": "Speichern",
      "addImage": "Bild hinzufügen*",
      "addAudio": "Audio aufnehmen",
      "maxPictureUpload": "Du kannst bis zu 3 Bilder hinzufügen",
      "requiredFields": "*Pflichtfelder",
      "errorMessage": "Dies ist ein Pflichtfeld.",
      "imageErrorMessage": "Du musst 1 bis 3 Bilder auswählen"
    },
    "list": {
      "edit": "Bearbeiten",
      "finished": "Fertig",
      "noWordsYet": "Du hast noch keine eigenen Wörter erstellt.",
      "create": "Eigenes Wort erstellen",
      "confirmDeletionPart1": "Willst du das Wort",
      "confirmDeletionPart2": "wirklich löschen?",
      "confirm": "Ja, löschen",
      "cancel": "Schliessen"
    }
  }
}
