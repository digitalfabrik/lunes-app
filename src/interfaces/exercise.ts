export interface IVocabularyOverviewScreen {
  navigation: any
  route: any
}

export interface IVocabularyTrainerScreen {
  navigation: any
  route: any
}

export interface IAlternativeWordProps {
  alt_word: string
  article: string
}

export interface IDocumentProps {
  id: number
  word: string
  article: string
  document_image: Array<{ id: number; image: string }>
  audio: string
  alternatives: IAlternativeWordProps[]
}

export interface IVocabularyOverviewListItemProps {
  id: number
  word: string
  article: string
  image: string
  audio: string
}

export interface IConfirmationModalProps {
  visible: boolean
  setIsModalVisible: Function
  navigation: any
  extraParams: any
}

export interface IAnswerSectionProps {
  tryLater: () => void
  currentDocumentNumber: number
  setCurrentDocumentNumber: Function
  documents: IDocumentProps[]
  finishExercise: Function
  trainingSet: string
  disciplineTitle: string
}

export interface IFeedbackProps {
  secondAttempt: boolean
  result: string
  document?: IDocumentProps
  input: string
}

export interface IActionsProps {
  tryLater: () => void
  giveUp: () => void
  result: string
  checkEntry: () => void
  getNextWord: () => void
  input: string
  isFinished: boolean
  secondAttempt: boolean
}
