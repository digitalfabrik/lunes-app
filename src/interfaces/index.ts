import { ReactElement } from 'react'

export interface ILoadingProps {
  children: ReactElement
  isLoading: boolean
}

export interface IButtonProps {
  onPress: () => void
  disabled?: boolean
  children: ReactElement
  theme?: string
  testID?: string
}
export interface IPopoverProps {
  setIsPopoverVisible: (param: boolean) => void
  isVisible: boolean
  children: ReactElement
}
