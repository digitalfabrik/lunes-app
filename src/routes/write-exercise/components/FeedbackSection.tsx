import React, { ReactElement } from 'react'
import { heightPercentageToDP as hp, widthPercentageToDP as wp } from 'react-native-responsive-screen'
import styled from 'styled-components/native'

import {
  AlmostCorrectFeedbackIcon,
  correct_background,
  CorrectFeedbackIcon,
  hint_background,
  incorrect_background,
  IncorrectFeedbackIcon
} from '../../../../assets/images'
import { SimpleResultType } from '../../../constants/data'
import { DocumentType } from '../../../constants/endpoints'
import labels from '../../../constants/labels.json'

const Background = styled.ImageBackground`
  width: ${wp(80)}px;
  height: ${hp(9)}px;
  margin-bottom: 40px;
  flex-direction: row;
  align-items: center;
  padding: 10px;
`

const StyledText = styled.Text`
  width: 100%;
  color: ${props => props.theme.colors.lunesBlack};
  padding: 0 20px 0 10px;
`

export interface FeedbackPropsType {
  result: SimpleResultType
  document: DocumentType
  submission: string | null
}

const Feedback = ({ result, document, submission }: FeedbackPropsType): ReactElement | null => {
  let Icon, background, message
  if (result === 'correct') {
    Icon = CorrectFeedbackIcon
    background = correct_background
    message = labels.exercises.write.feedback.correct
  } else if (result === 'similar' && submission) {
    Icon = AlmostCorrectFeedbackIcon
    background = hint_background
    message = `${labels.exercises.write.feedback.almostCorrect1} „${submission}“ ${labels.exercises.write.feedback.almostCorrect2}`
  } else {
    Icon = IncorrectFeedbackIcon
    background = incorrect_background
    message = `${labels.exercises.write.feedback.wrong} „${document.article.value} ${document.word}“`
  }

  return (
    <Background source={background} testID='background-image'>
      <Icon width={28} height={28} />
      <StyledText numberOfLines={2} ellipsizeMode='tail' testID={'feedback-write-exercise'}>
        {message}
      </StyledText>
    </Background>
  )
}

export default Feedback
