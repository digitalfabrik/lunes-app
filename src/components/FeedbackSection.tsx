import React from 'react'
import { View, Text, ImageBackground, StyleSheet } from 'react-native'
import { IFeedbackProps } from '../interfaces/exercise'
import {
  CorrectFeedbackIcon,
  IncorrectFeedbackIcon,
  AlmostCorrectFeedbackIcon,
  incorrect_background,
  hint_background,
  correct_background
} from '../../assets/images'
import { ARTICLES } from '../constants/data'
import { COLORS } from '../constants/colors'
import { widthPercentageToDP as wp, heightPercentageToDP as hp } from 'react-native-responsive-screen'

export const styles = StyleSheet.create({
  messageContainer: {
    width: wp('80%'),
    height: hp('9%'),
    flexDirection: 'row',
    alignItems: 'center',
    marginTop: -hp('4%'),
    marginBottom: hp('3%')
  },
  imageBackground: {
    width: wp('80%'),
    height: hp('9%'),
    flexDirection: 'row',
    alignItems: 'center',
    padding: 10
  },
  textContainer: {
    marginHorizontal: 5,
    paddingRight: 15,
    paddingLeft: 5
  },
  text: {
    fontSize: wp('3.5%'),
    fontFamily: 'SourceSansPro-Regular',
    fontWeight: 'normal',
    color: COLORS.lunesBlack
  }
})

const Feedback = ({ result, document, input, secondAttempt }: IFeedbackProps) => {
  const Icon =
    result === 'correct'
      ? CorrectFeedbackIcon
      : result === 'incorrect' || !secondAttempt
      ? IncorrectFeedbackIcon
      : AlmostCorrectFeedbackIcon

  const background =
    result === 'correct'
      ? correct_background
      : result === 'incorrect' || !secondAttempt
      ? incorrect_background
      : hint_background

  const message =
    result === 'correct'
      ? 'Great, keep it up! \nThe Word you filled in is correct.'
      : result === 'incorrect' || !secondAttempt
      ? `What a pity! Your entry is incorrect,\nthe correct answer is: ${
          document?.article?.toLowerCase() === ARTICLES.diePlural ? 'die' : document?.article
        } ${document?.word}`
      : `Your entry ${input} is almost correct. Check for upper and lower case.`

  return result !== '' || secondAttempt ? (
    <View style={styles.messageContainer}>
      <ImageBackground source={background} style={styles.imageBackground} testID='background-image'>
        <Icon width={28} height={28} />
        <View style={styles.textContainer}>
          <Text numberOfLines={2} ellipsizeMode='tail' style={styles.text}>
            {message}
          </Text>
        </View>
      </ImageBackground>
    </View>
  ) : null
}

export default Feedback
