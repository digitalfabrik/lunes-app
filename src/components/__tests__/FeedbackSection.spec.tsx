import 'react-native'
import React from 'react'
import FeedbackSection from '../FeedbackSection'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { IFeedbackProps } from '../../interfaces/exercise'
import { ARTICLES } from '../../constants/data'
import { CorrectFeedbackIcon, IncorrectFeedbackIcon, AlmostCorrectFeedbackIcon } from '../../../assets/images'

describe('Components', () => {
  describe('Feedback section', () => {
    const defaultFeedbackProps: IFeedbackProps = {
      input: '',
      result: '',
      secondAttempt: false,
      document: {
        alternatives: [],
        article: '',
        audio: '',
        id: 0,
        image: '',
        word: ''
      }
    }

    it('should render without issues', () => {
      const component = shallow(<FeedbackSection {...defaultFeedbackProps} />)
      expect(toJson(component)).toMatchSnapshot()
    })

    it('should not render when there is no result and secondAttempt is false', () => {
      const feedbackProps: IFeedbackProps = {
        ...defaultFeedbackProps
      }

      const component = shallow(<FeedbackSection {...feedbackProps} />)
      expect(component.isEmptyRender()).toBe(true)
    })

    it('should render success feedback when the result is correct', () => {
      const feedbackProps: IFeedbackProps = {
        ...defaultFeedbackProps,
        result: 'correct'
      }
      const message = 'Great, keep it up! \nThe Word you filled in is correct.'
      const icon = <CorrectFeedbackIcon width={28} height={28} />

      const component = shallow(<FeedbackSection {...feedbackProps} />)
      expect(component.find('Text').props().children).toBe(message)
      expect(component.contains(icon)).toBe(true)
    })

    it('should render error feedback when the result is incorrect and the second attempt is false', () => {
      const feedbackProps: IFeedbackProps = {
        ...defaultFeedbackProps,
        result: 'incorrect',
        document: {
          article: 'der',
          word: 'Winkel',
          audio: '',
          alternatives: [],
          id: 1,
          image: ''
        }
      }
      const message = `What a pity! Your entry is incorrect,\nthe correct answer is: ${
        feedbackProps.document?.article?.toLowerCase() === ARTICLES.diePlural ? 'die' : feedbackProps.document?.article
      } ${feedbackProps.document?.word}`
      const icon = <IncorrectFeedbackIcon width={28} height={28} />

      const component = shallow(<FeedbackSection {...feedbackProps} />)
      expect(component.find('Text').props().children).toBe(message)
      expect(component.contains(icon)).toBe(true)
    })

    it('should render almost correct feedback when the result is similar', () => {
      const feedbackProps: IFeedbackProps = {
        ...defaultFeedbackProps,
        result: 'similar',
        input: 'der winkel',
        secondAttempt: true
      }
      const message = `Your entry ${feedbackProps.input} is almost correct. Check for upper and lower case.`
      const icon = <AlmostCorrectFeedbackIcon width={28} height={28} />

      const component = shallow(<FeedbackSection {...feedbackProps} />)
      expect(component.find('Text').props().children).toBe(message)
      expect(component.contains(icon)).toBe(true)
    })
  })
})
