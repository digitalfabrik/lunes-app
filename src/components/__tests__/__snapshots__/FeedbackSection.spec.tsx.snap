// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Feedback section should render without issues 1`] = `""`;
